bot/bot.py

```diff
@@ -137,10 +130,73 @@ def contract_roll( num_dice, difficulty=6, exert=False, label_text=None):
 	return "\n".join(response)
 
 
+class CommandHelper:
+	def __init__(self, dice_diff, num_dice, difficulty):
+		@commands.command(name=dice_diff)
+		async def d_command(ctx, *args):
+			await dice_roll(ctx, num_dice, difficulty, args)
+		self.command = d_command
+
+async def dice_roll(ctx, dice, diff, args):
+	await roll_dice(ctx, dice, diff, args)
+
+def create_roll_combinations(dice_bot):
+	min_dice = 1
+	max_dice = 50
+	min_diff = 1
+	max_diff = 15
+
+	for dice in range(min_dice,max_dice):
+		dice_diff = "!{}".format(dice)
+		default_difficulty = 6
+		obj = CommandHelper(dice_diff, dice, default_difficulty)
+		dice_bot.add_command(obj.command)
+
+		for diff in range(min_diff,max_diff):
+			dice_diff = "!{}@{}".format(dice, diff)
+			obj = CommandHelper(dice_diff, dice, diff)
+			dice_bot.add_command(obj.command)
+
 intents = discord.Intents.default()
 intents.message_content = True
 bot = commands.Bot(command_prefix='!', intents=intents)
-bot.add_command(roll_dice)
 bot.add_command(roll_low)
 bot.add_command(roll_high)
+bot.add_command(call_for_help)
+create_roll_combinations(bot)
+
+@bot.event
+async def on_command_error(ctx, error):
+	command_from_user = ctx.invoked_with
+	commands_list = list(command_from_user)
+	if commands_list[0] != '!':
+		return
+
+	contextual_error = ""
+	if not commands_list[1].isdigit():
+		contextual_error = contextual_error + "\nnumber of dice expected, input not a valid number"
```

Comment:
Capitalize the start of sentences and put periods. E.g. `\nNumber of dice expected. Input is not a valid number.`

Reply:
Same goes for all these usage messages.